it/controllers/errors/AddressHasChangedPageControllerISpec.scala

```diff
@@ -0,0 +1,167 @@
+    }
+
+    "render correctly when the user is an agent" should {
+      val request = FakeRequest("GET", urlPath).withHeaders(headers: _*)
+
+      lazy val result: Future[Result] = {
+        authoriseAgent()
+        route(app, request).get
+      }
+
+      implicit def document: () => Document = () => Jsoup.parse(Helpers.contentAsString(result))
+
+      "returns status of OK(200)" in {
+        status(result) shouldBe OK
+      }
+
+      welshToggleCheck("English")
+      titleCheck(titleAgent)
+      h1Check(headingAgent, "xl")
+      textOnPageCheck(addressHasChangedTextAgent, addressHasChangedSelector)
+      textOnPageCheck(submitYourReturnAgent, submitYourReturnSelector)
+      textOnPageCheck(incomeTaxReturnButtonText, incomeTaxReturnButtonSelector)
+      buttonCheck(incomeTaxReturnButtonText, incomeTaxReturnButtonSelector, Some(incomeTaxReturnButtonLink))
+    }
+  }
+
+  "Rendering the address change error page in Welsh" should {
+
+    val headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear), "Csrf-Token" -> "nocheck", HeaderNames.ACCEPT_LANGUAGE -> "cy")
+
+    "render correctly when the user is an individual" should {
+      val request = FakeRequest("GET", urlPath).withHeaders(headers: _*)
+
+      lazy val result: Future[Result] = {
+        authoriseIndividual()
+        route(app, request).get
+      }
+
+      implicit def document: () => Document = () => Jsoup.parse(Helpers.contentAsString(result))
+
+      "returns status of OK(200)" in {
+        status(result) shouldBe OK
+      }
+
+      welshToggleCheck("Welsh")
+      titleCheck(titleIndividual)
+      h1Check(headingIndividual, "xl")
+      textOnPageCheck(addressHasChangedTextIndividual, addressHasChangedSelector)
+      textOnPageCheck(submitYourReturnIndividual, submitYourReturnSelector)
+      textOnPageCheck(incomeTaxReturnButtonText, incomeTaxReturnButtonSelector)
+      buttonCheck(incomeTaxReturnButtonText, incomeTaxReturnButtonSelector, Some(incomeTaxReturnButtonLink))
+    }
+
+    "render correctly when the user is an agent" should {
+      val request = FakeRequest("GET", urlPath).withHeaders(headers: _*)
+
+      lazy val result: Future[Result] = {
+        authoriseAgent()
+        route(app, request).get
+      }
+
+      implicit def document: () => Document = () => Jsoup.parse(Helpers.contentAsString(result))
+
+      "returns status of OK(200)" in {
+        status(result) shouldBe OK
+      }
+
+      welshToggleCheck("Welsh")
+      titleCheck(titleAgent)
+      h1Check(headingAgent, "xl")
+      textOnPageCheck(addressHasChangedTextAgent, addressHasChangedSelector)
+      textOnPageCheck(submitYourReturnAgent, submitYourReturnSelector)
+      textOnPageCheck(incomeTaxReturnButtonText, incomeTaxReturnButtonSelector)
+      buttonCheck(incomeTaxReturnButtonText, incomeTaxReturnButtonSelector, Some(incomeTaxReturnButtonLink))
+    }
+  }
+
+}
```

Comment:
I usually don't mention the above but could we get some new lines here and in the messages files please?